homeassistant/components/matter/entity.py

```diff
@@ -115,10 +115,28 @@ async def async_will_remove_from_hass(self) -> None:
                 # suppress ValueError to prevent race conditions
                 unsub()
 
+    async def async_update(self) -> None:
+        """Call when the entity needs to be updated."""
+        if not self._attr_should_poll:
+            return
+        # manually poll/refresh the primary value
+        await self.matter_client.refresh_attribute(
+            self._endpoint.node.node_id,
+            self.get_matter_attribute_path(self._entity_info.primary_attribute),
+        )
+        self._update_from_device()
+        self.async_write_ha_state()
```

Comment:
This is not needed. `async_write_ha_state` is called after we call `async_update`. https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L680

Reply:
ah... now I get what you mean... good morning.